src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,265 @@
+        }
+
+        public virtual IEnumerator<TToken> TokensOf(StoreView snapshot, UInt160 owner)
+        {
+            return snapshot.Storages.Find(CreateStorageKey(Prefix_Ownership, owner).ToArray()).Select(p =>
+            {
+                UInt160 innerKey = new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+                return snapshot.Storages.TryGet(CreateTokenKey(innerKey)).GetInteroperable<TToken>();
+            }).GetEnumerator();
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.InteropInterface, CallFlags.None, ParameterTypes = new[] { ContractParameterType.ByteArray }, ParameterNames = new[] { "tokenId" })]
+        public StackItem OwnerOf(ApplicationEngine engine, Array args)
+        {
+            byte[] tokenId = args[0].GetSpan().ToArray();
+            if (tokenId.Length > MaxTokenIdLength) return false;
+            return new InteropInterface(OwnerOf(engine.Snapshot, tokenId));
+        }
+
+        public virtual IEnumerator<UInt160> OwnerOf(StoreView snapshot, byte[] tokenId)
+        {
+            UInt160 innerKey = GetInnerKey(tokenId);
+            return snapshot.Storages.Find(CreateStorageKey(Prefix_Ownership, innerKey).ToArray()).Select(p =>
+            {
+                return new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+            }).GetEnumerator();
+        }
+
+        [ContractMethod(0_08000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.Hash160, ContractParameterType.ByteArray }, ParameterNames = new[] { "to", "tokenId" })]
+        public virtual StackItem Transfer(ApplicationEngine engine, Array args)
+        {
+            UInt160 to = new UInt160(args[0].GetSpan());
+            byte[] tokenId = args[1].GetSpan().ToArray();
+            if (tokenId.Length > MaxTokenIdLength) return false;
+            return Transfer(engine, to, tokenId);
+        }
+
+        public virtual bool Transfer(ApplicationEngine engine, UInt160 to, byte[] tokenId)
+        {
+            IEnumerator<UInt160> enumerator = OwnerOf(engine.Snapshot, tokenId);
+            if (!enumerator.MoveNext()) return false;
+            UInt160 owner = enumerator.Current;
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, owner)) return false;
+
+            var snapshot = engine.Snapshot;
+            ContractState contract_to = snapshot.Contracts.TryGet(to);
+            if (contract_to?.Payable == false) return false;
+
+            UInt160 innerKey = GetInnerKey(tokenId);
+            snapshot.Storages.Delete(CreateOwnershipKey(owner, innerKey));
+            snapshot.Storages.Delete(CreateOwnershipKey(innerKey, owner));
+            snapshot.Storages.Add(CreateOwnershipKey(to, innerKey), new StorageItem(new TAccount() { Balance = Factor }));
+            snapshot.Storages.Add(CreateOwnershipKey(innerKey, to), new StorageItem(new byte[0]));
```

Comment:
They share the same prefixes? Maybe using different prefixes can improve the performance.

Reply:
Yes, they are use the same prefixes. I have splited into two different prefixes now.